runtime/compiler/env/j9method.cpp

```diff
@@ -6757,6 +6757,39 @@ TR_ResolvedJ9Method::getResolvedDynamicMethod(TR::Compilation * comp, I_32 callS
 #else
    TR_ResolvedMethod *result = NULL;
 
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+   TR_ResolvedMethod * owningMethod = comp->getCurrentMethod();
+   J9InvokeCacheEntry *invokeCache = (J9InvokeCacheEntry *) owningMethod->callSiteTableEntryAddress(callSiteIndex);
+   if (!invokeCache)
+      comp->failCompilation<TR::CompilationException>("unable to retrieve side table entry for invokeDynamic");
+      {
+      TR::VMAccessCriticalSection getResolvedDynamicMethod(fej9());
```

Comment:
Reading constant pool entries does require a VM access. I think the VM access including the original one can be removed.